src/features/process_dataset.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# ------------------------------
+# File       : process_dataset.py
+# Created by : Yanis De Busschere, Luka Secilmis, Thomas Ecabert
+"""Module/script used to process the raw dataset."""
+
+import logging
+import sys
+from math import ceil
+
+from bs4 import BeautifulSoup
+import pandas as pd
+
+from src.utils.file_utils import (
+    PROCESSED_DATASETS_EXTENSION,
+    PROCESSED_DATASETS_PREFIX,
+    RAW_DATASETS_PATH,
+    PROCESSED_DATASETS_PATH,
+    extract_date,
+    find_latest_raw_dataset,
+)
+
+from src.utils.init_utils import init_logger, init_seed, SEED
+
+KEPT_FIELDS = ["description", "spam"]
+CHUNK_SIZE = 100000
+
+
+def parse_description(description: str) -> str:
+    """Cleans up the description by removing HTML tags.
+
+    Args:
+        description (str): Description to clean up.
+
+    Returns:
+        str: Cleaned up description.
+    """
+    return BeautifulSoup(description, "html.parser").get_text()
+
+
+def process_dataset(raw_dataset_path: str, processed_dataset_path: str) -> None:
+    """Process the raw dataset, process it and save it.
+
+    Args:
+        raw_dataset_path (str): Path to the raw dataset.
+        processed_dataset_path (str): Path to the processed dataset.
+    """
+
+    spams = pd.DataFrame()
+    hams = pd.DataFrame()
+
+    logging.info(f'Processing dataset from "{raw_dataset_path}".')
+    logging.info(f'Processed dataset will be saved to "{processed_dataset_path}".')
+
+    nb_lines = 0
+    with open(raw_dataset_path, "r") as f:
+        for _ in f:
+            nb_lines += 1
+    chunk_done = 0
+
+    for chunk in pd.read_json(raw_dataset_path, lines=True, chunksize=CHUNK_SIZE):
+        logging.info(
+            f"Processing chunk {chunk_done + 1} out of {ceil(nb_lines / CHUNK_SIZE)}."
+        )
+
+        chunk = chunk[KEPT_FIELDS].dropna()
+
+        chunk_spams = chunk[chunk["spam"] == True]
+        chunk_spams["description"] = chunk_spams["description"].map(parse_description)
```

Comment:
Using `.map()` here could be parallelized to make use of mult{threading,processing}.